File: Instagram1/app/src/main/java/com/example/instagram1/MyAdapter.java
package com.example.instagram1;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import java.util.List;

public class MyAdapter extends RecyclerView.Adapter<MyAdapter.ViewHolder> {
    private int[] images;
    public MyAdapter(int[] images)
    {
        this.images=images;
    }
    private List<Listitem> listItems;
    private Context context;

    public MyAdapter(List<Listitem> listItems, Context context) {
        this.listItems = listItems;
        this.context = context;
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.list_item,parent,false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        int image_id = images[position];
        holder.post_image.setImageResource(image_id);
        holder.tvLikes.setText("5"+position+"000");
    }

    @Override
    public int getItemCount() {
        return images.length;
    }

    public class ViewHolder extends RecyclerView.ViewHolder{
        ImageView post_image;
        TextView tvLikes;


        public ViewHolder(@NonNull View itemView) {
            super(itemView);
            post_image= itemView.findViewById(R.id.post_image);
            tvLikes = itemView.findViewById(R.id.likes_number);
        }
    }



}

File: Insta_kotlin/app/src/main/java/com/example/instagramkotlin/MyAdapter.kt
package com.example.instagramkotlin

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import kotlinx.android.synthetic.main.listitem.view.*
import java.util.*
import com.bumptech.glide.Glide
import com.bumptech.glide.request.RequestOptions

class MyAdapter : RecyclerView.Adapter<RecyclerView.ViewHolder>() {
    private var items: List<ListItem> = ArrayList()
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerView.ViewHolder {
        return ItemViewHolder(
            LayoutInflater.from(parent.context).inflate(R.layout.listitem, parent, false)
        )
    }

    override fun onBindViewHolder(holder: RecyclerView.ViewHolder, position: Int) {
        when (holder) {
            is ItemViewHolder -> {
                holder.run { bind(items.get(position)) }
            }
        }
    }

    override fun getItemCount() = items.size

    fun submitList(itemList: List<ListItem>) {
        items = itemList
    }

    class ItemViewHolder constructor(
        itemView: View
    ) : RecyclerView.ViewHolder(itemView) {
        val userImage = itemView.user_image
        val userName = itemView.user_name
        val likesNumber = itemView.likes_number

        fun bind(listItem: ListItem) {
            userName.text = listItem.username
            likesNumber.run { setText(listItem.likes) }
            val requestOptions = RequestOptions()
                .placeholder(R.drawable.ic_launcher_background)
                .error(R.drawable.ic_launcher_background)
            Glide.with(itemView.context)
                .applyDefaultRequestOptions(requestOptions)
                .load(listItem.image)
                .into(userImage)
        }
    }

}